wumpus.py

```diff
@@ -103,7 +107,54 @@ def _breeze_stench_rules(self):
         return clauses
 
 
-def create_wumpus_world():
+def create_wumpus_world(map_width=4, map_height=4):
+    grid = [[set() for i in map_width] for j in map_height]
+    # can write a function for this type of random coord generation (nothing should be on 0,0)
+    wumpus_location = randrange(1, map_width * map_height)
+    wumpus_y = int(wumpus_location / map_width)
+    wumpus_x = wumpus_location - (wumpus_y * map_width)
+    grid[wumpus_y][wumpus_x].add(set([Property.WUMPUS]))
+    # can maybe use breeze stench rules to simplify this? or abstract into a function
+    if wumpus_x > 0:
```

Comment:
Hi, you can even create a class `WumpusWorldGenerator` (or something like that), and add internal methods for this checks. For example:

```python
def _adjacent_cords(self, x, y):
    dirs = [(0, 1), (1, 0), (0, -1), (-1, 0)]
    for dx, dy in dirs:
        xnew, ynew = x - dx, y - dx
        if self._valid_coord(xnew, ynew):
            yield xnew, ynew

def _valid_coord(self, x, y) -> bool:
    return (x >= 0 and x < self.map_width and y >= 0 and y < self.map_height)
```